tinygrad/runtime/ops_webgpu.py

```diff
@@ -0,0 +1,133 @@
+        else:
+          if args[1] == "global" and pend_close:
+            depth -= 1
+            kk(pend_close) 
+            pend_close = None
+          depth -= 1
+          kk("}"*len(args[0])  + f" /* {args[1]} */")
+      elif uop == UOps.LOAD and newvar is not None:
+        if self.bufs[args.i] is not None and isinstance(self.bufs[args.i].realized, RawConst):
+          assert newvar.dtype == dtypes.float, "only floats"
+          val = self.float_const(self.bufs[args.i].realized._buf)
+        else:
+          val = f"{bufnames[args.i]}[{args.idx.render(render_cl)}]"
+        if args.valid.min == 1: kk(f"let {newvar.render()} = f32({val});")
+        elif args.valid.render(render_cl) == "0": kk(f"var {newvar.render()} = 0.0f;")
+        else: kk(f"var {newvar.render()} = select(0.0f, f32({val}), bool({args.valid.render(render_cl)}));")
+      elif uop == UOps.ALU:
+        assert newvar is not None
+        if newvar in vin: kk(f"{newvar.render()} = {code_for_op[args](*[x.render() for x in vin])};")
+        else: kk(f"let {newvar.render()} = {code_for_op[args](*[x.render() for x in vin])};")
+      elif uop == UOps.STORE:
+        val = vin[0].render()
+        if vin[0].dtype != self.bufs[args.i].dtype:
+          val = f"{type_map[self.bufs[args.i].dtype]}({val})"
+        kk(f"{bufnames[args.i]}[{args.idx.render(render_cl)}] = {val};")
+      elif uop == UOps.CONST:
+        assert newvar is not None
+        kk(f"var {newvar.render()} = {self.float_const(args)};")
+      elif uop == UOps.DEFINE_LOCAL: kk(f"var {args[0]} = array<f32,{args[1]}>();")
+      elif uop == UOps.BARRIER: kk("workgroupBarrier();")
+      else: raise RuntimeError(f"failed to render {uop}")
+    # Function name itself isn't unique
+    assert all(x <= 65535 for x in global_size), "WEBGPU max global size is 65535 in any dimension"
+    assert len([x for x in self.bufs if not isinstance(x, LocalBuffer) and not isinstance(x.realized, RawConst)]) <= 31, "WEBGPU max number of buffers is 31" 
+    function_name = f"{self.function_name}_{abs(hash(self.key))}"
+    bind_it = iter(range(len(self.bufs)))
+    prg = "\n".join([f"@group(0) @binding({next(bind_it)}) var<storage,read_write> data{i}: array<{type_map[x.dtype]}>;" for i,x in enumerate(self.bufs) if not isinstance(x, LocalBuffer) and not isinstance(x.realized, RawConst)])
+    prg += f"\n@compute @workgroup_size(1) fn {function_name}(@builtin(global_invocation_id) gindex: vec3<u32>, @builtin(local_invocation_id) lindex: vec3<u32>) {{\n" + "\n".join(kernel) + "\n}" # TODO: revert local_size {','.join([str(x) for x in local_size])} once bug is fixed
+    return ASTRunner(function_name, prg, global_size[::-1] if len(global_size) else [1], local_size[::-1] if len(local_size) else [1])
+
+class RawWebGPUBuffer(RawBufferCopyIn):
+  def __init__(self, size:int, dtype:DType):
+    assert dtype not in [dtypes.int8,dtypes.uint8,dtypes.int64,dtypes.uint64,dtypes.float64], f"dtype {dtype} not supported on WEBGPU"
```

Comment:
could you use arbitrary float https://www.w3.org/TR/WGSL/#example-bbb80169 for float64?